net/usrsock/usrsock_conn.c

```diff
@@ -94,23 +97,36 @@ static void _usrsock_semgive(FAR sem_t *sem)
 FAR struct usrsock_conn_s *usrsock_alloc(void)
 {
   FAR struct usrsock_conn_s *conn;
+#ifdef CONFIG_NET_ALLOC_CONNS
+  int i;
+#endif
 
   /* The free list is protected by a semaphore (that behaves like a mutex). */
 
   _usrsock_semtake(&g_free_sem);
+#ifdef CONFIG_NET_ALLOC_CONNS
+  if (dq_peek(&g_free_usrsock_connections) == NULL)
+    {
+      conn = kmm_zalloc(sizeof(*conn) * CONFIG_NET_USRSOCK_CONNS);
+      if (conn != NULL)
+        {
+          for (i = 0; i < CONFIG_NET_USRSOCK_CONNS; i++)
+            {
+              dq_addlast(&conn[i].node, &g_free_usrsock_connections);
+            }
+        }
+    }
+#endif
```

Comment:
Why not to do allocation in `usrsock_initialize`?

Reply:
allocate on use, all conn instances will be obtained through the allocator if g_free_usrsock_connections is NULL

Reply:
In such case we do not limit the allocated connection number by `CONFIG_NET_ALLOC_CONNS`. The current condition seems to be true for the first time and when `CONFIG_NET_ALLOC_CONNS` number of connections are allocated.

Reply:
So why we are not allocating one by one, but with `CONFIG_NET_ALLOC_CONNS` portions?